app/services/rss_notifications_checker.rb

```diff
@@ -0,0 +1,48 @@
+require 'rss'
+require 'date'
+
+class RssNotificationsChecker
```

Comment:
this file is in the wrong place, it should be in app/services/ui_notifications/.

Also, it should inherit from the notification base class imho.